### ISSUE TITLE
Uncaught TypeError: Cannot read property 'marker' of undefined

### ISSUE THREAD

**Issue by marcoslhc** [t9md/atom-vim-mode-plus#7]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.0.19
**System**: Mac OS X 10.10.5
**Thrown From**: [vim-mode-plus](https://github.com/t9md/atom-vim-mode-plus) package, v0.1.3

### Stack Trace

Uncaught TypeError: Cannot read property 'marker' of undefined

```
At /Users/marcoslh/.atom/packages/vim-mode-plus/lib/vim-state.coffee:159

TypeError: Cannot read property 'marker' of undefined
  at BlockwiseMoveDown.VisualBlockwise.isReversed (/Users/marcoslh/.atom/packages/vim-mode-plus/lib/visual-blockwise.coffee:35:17)
  at BlockwiseMoveDown.isForward (/Users/marcoslh/.atom/packages/vim-mode-plus/lib/visual-blockwise.coffee:78:10)
  at BlockwiseMoveDown.execute (/Users/marcoslh/.atom/packages/vim-mode-plus/lib/visual-blockwise.coffee:84:9)
  at OperationStack.module.exports.OperationStack.process (/Users/marcoslh/.atom/packages/vim-mode-plus/lib/operation-stack.coffee:80:10)
  at /Users/marcoslh/.atom/packages/vim-mode-plus/lib/operation-stack.coffee:33:8
  at OperationStack.module.exports.OperationStack.withLock (/Users/marcoslh/.atom/packages/vim-mode-plus/lib/operation-stack.coffee:38:7)
  at OperationStack.module.exports.OperationStack.push (/Users/marcoslh/.atom/packages/vim-mode-plus/lib/operation-stack.coffee:32:6)
  at atom-text-editor.commands.(anonymous function) (/Users/marcoslh/.atom/packages/vim-mode-plus/lib/vim-state.coffee:156:29)
  at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/command-registry.js:245:29)
  at /opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61
  at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:524:16)
  at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:347:22)
  at HTMLDocument.module.exports.WindowEventHandler.onKeydown (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:184:20)

```

### Commands

```
     -0:38.3.0 vim-mode-plus:move-down (atom-text-editor.editor.vim-mode-plus.reversed.is-focused.normal-mode)
  7x -0:37.7.0 vim-mode-plus:move-up (atom-text-editor.editor.vim-mode-plus.reversed.is-focused.normal-mode)
  2x -0:35.8.0 vim-mode-plus:move-down (atom-text-editor.editor.vim-mode-plus.reversed.is-focused.normal-mode)
  3x -0:35.1.0 vim-mode-plus:move-up (atom-text-editor.editor.vim-mode-plus.reversed.is-focused.normal-mode)
  3x -0:34.3.0 vim-mode-plus:move-left (atom-text-editor.editor.vim-mode-plus.reversed.is-focused.normal-mode)
     -0:31.3.0 vim-mode-plus:activate-blockwise-visual-mode (atom-text-editor.editor.vim-mode-plus.reversed.is-focused.normal-mode)
  5x -0:30.5.0 vim-mode-plus:blockwise-move-down (atom-text-editor.editor.vim-mode-plus.reversed.is-focused.visual-mode.blockwise)
 20x -0:28.7.0 vim-mode-plus:move-right (atom-text-editor.editor.vim-mode-plus.is-focused.visual-mode.blockwise)
     -0:26 vim-mode-plus:move-left (atom-text-editor.editor.vim-mode-plus.is-focused.visual-mode.blockwise)
     -0:19.8.0 vim-mode-plus:change (atom-text-editor.editor.vim-mode-plus.is-focused.visual-mode.blockwise)
     -0:16.5.0 vim-mode-plus:activate-normal-mode (atom-text-editor.editor.vim-mode-plus.is-focused.insert-mode.autocomplete-active)
     -0:16.2.0 vim-mode-plus:undo (atom-text-editor.editor.vim-mode-plus.is-focused.normal-mode)
 20x -0:14.3.0 vim-mode-plus:move-left (atom-text-editor.editor.vim-mode-plus.is-focused.normal-mode)
     -0:05.8.0 vim-mode-plus:activate-blockwise-visual-mode (atom-text-editor.editor.vim-mode-plus.is-focused.normal-mode)
 13x -0:05.2.0 vim-mode-plus:move-right (atom-text-editor.editor.vim-mode-plus.is-focused.visual-mode.blockwise)
  2x -0:02.6.0 vim-mode-plus:blockwise-move-down (atom-text-editor.editor.vim-mode-plus.is-focused.visual-mode.blockwise)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "symbols-view"
    ],
    "themes": [
      "seti-ui",
      "base16-tomorrow-dark-theme"
    ]
  }
}
```

### Installed Packages

```coffee
# User
atom-ctags, v4.3.0
atom-django, v0.3.0
color-picker, v2.0.12
django-templates, v0.5.0
editorconfig, v1.2.0
emmet, v2.3.12
git-log, v0.4.1
git-tab-status, v1.9.2
language-matlab, v0.1.1
linter, v1.6.0
linter-flake8, v1.6.1
linter-jscs, v3.0.0
linter-jshint, v1.2.0
linter-python-pep8, v0.2.0
node-debugger, v1.1.8
pigments, v0.15.0
pretty-json, v0.4.1
remote-sync, v3.1.2
seti-ui, v0.8.0
symbols-tree-view, v0.10.0
vim-mode-plus, v0.1.3

# Dev
No dev packages
```

**Comment by t9md** [comment]
Thanks for reporting this.
I make sure that Tail and Head is ALWAYS available in visual-block mode.
Although I can choose different fix approach guard tail reference like `getTail().?` but I want to avoid this approach since its hide true cause.

I believe I can close this, if you have same issue again, reopen or report another issue.

**Comment by averrin** [comment]
Atom: 1.11-beta5

Its happened after operation with persistantSelection, but i cant reproduce it.

/home/user/.atom/packages/vim-mode-plus/lib/operation-stack.coffee:117

TypeError: Cannot read property 'marker' of undefined
    at /home/user/.atom/packages/vim-mode-plus/lib/operator.coffee:97:87
    at Function.module.exports.Emitter.simpleDispatch (/usr/share/atom-beta/resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/usr/share/atom-beta/resources/app.asar/node_modules/event-kit/lib/emitter.js:129:28)
    at OperationStack.finish (/home/user/.atom/packages/vim-mode-plus/lib/operation-stack.coffee:156:23)
    at OperationStack.execute (/home/user/.atom/packages/vim-mode-plus/lib/operation-stack.coffee:147:8)
    at OperationStack.process (/home/user/.atom/packages/vim-mode-plus/lib/operation-stack.coffee:130:8)
    at OperationStack.run (/home/user/.atom/packages/vim-mode-plus/lib/operation-stack.coffee:95:10)
    at atom-text-editor.<anonymous> (/home/user/.atom/packages/vim-mode-plus/lib/base.coffee:326:33)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/usr/share/atom-beta/resources/app.asar/src/command-registry.js:260:29)
    at /usr/share/atom-beta/resources/app.asar/src/command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/usr/share/atom-beta/resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:580:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/usr/share/atom-beta/resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:388:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeyEvent (/usr/share/atom-beta/resources/app.asar/src/window-event-handler.js:114:36)
    at HTMLDocument.<anonymous> (/usr/share/atom-beta/resources/app.asar/src/window-event-handler.js:3:61)

**Comment by t9md** [comment]
Thanks for reporting, released patched v0.58.2.
